src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -215,51 +256,30 @@ const DomainFragment = ({
         unfollowingDomains.push(item);
       }
     });
-    if (isFirstTimeOpen)
-      return [
-        followingDomains.map((item, index) =>
-          __renderDiscoveryItem(
-            FROM_FOLLOWED_DOMAIN_INITIAL,
-            'followedDomainDiscovery',
-            {...item, user_id_follower: item.user_id_follower},
-            index
-          )
-        )
-      ]
-        .concat([
-          route.name !== 'Followings' && (
-            <DiscoveryTitleSeparator text="Suggested Domains" key="domain-title-separator" />
-          )
-        ])
-        .concat(
-          unfollowingDomains.map((item, index) =>
-            __renderDiscoveryItem(
-              FROM_FOLLOWED_DOMAIN_INITIAL,
-              'followedDomainDiscovery',
-              {...item, user_id_follower: item.user_id_follower},
-              index + followingDomains.length
-            )
-          )
-        );
+    const data = isFirstTimeOpen
+      ? [
+          ...followingDomains.map((item, index) => ({
+            ...item,
+            user_id_follower: item.user_id_follower
+          })),
+          {separator: true},
+          ...unfollowingDomains.map((item, index) => ({
+            ...item,
+            user_id_follower: item.user_id_follower
+          }))
+        ]
+      : [...newMapFollowedDomain, {separator: true}, ...newMapUnfollowedDomain];
 
     return (
-      <>
-        {newMapFollowedDomain.map((item, index) =>
-          __renderDiscoveryItem(FROM_FOLLOWED_DOMAIN, 'followedDomainDiscovery', item, index)
-        )}
-
-        {route.name !== 'Followings' &&
-          unfollowedDomains.length > 0 &&
-          followedDomains.length > 0 && (
-            <View style={styles.unfollowedHeaderContainer}>
-              <Text style={styles.unfollowedHeaders}>{StringConstant.discoveryMoreDomains}</Text>
-            </View>
-          )}
-        {route.name !== 'Followings' &&
-          newMapUnfollowedDomain.map((item, index) =>
-            __renderDiscoveryItem(FROM_UNFOLLOWED_DOMAIN, 'unfollowedDomainDiscovery', item, index)
-          )}
-      </>
+      <FlatList
+        onMomentumScrollBegin={handleScroll}
+        contentContainerStyle={{paddingBottom: 100}}
+        data={data}
+        keyExtractor={(_, index) => index.toString()}
+        renderItem={renderItemList}
+        onEndReached={() => fetchData()}
+        onEndReachedThreshold={0.6}
+      />
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `FlatList` for rendering the list of domains is a good improvement. It provides better performance for long lists because it only renders items that are currently visible and doesn't render all items at once.
```diff
- if (isFirstTimeOpen)
-   return [
-     followingDomains.map((item, index) =>
-       __renderDiscoveryItem(
-         FROM_FOLLOWED_DOMAIN_INITIAL,
-         'followedDomainDiscovery',
-         {...item, user_id_follower: item.user_id_follower},
-         index
-       )
-     )
-   ]
-     .concat([
-       route.name !== 'Followings' && (
-         <DiscoveryTitleSeparator text="Suggested Domains" key="domain-title-separator" />
-       )
-     ])
-     .concat(
-       unfollowingDomains.map((item, index) =>
-         __renderDiscoveryItem(
-           FROM_FOLLOWED_DOMAIN_INITIAL,
-           'followedDomainDiscovery',
-           {...item, user_id_follower: item.user_id_follower},
-           index + followingDomains.length
-         )
-       )
-     );
- 
-  return (
-   <>
-     {newMapFollowedDomain.map((item, index) =>
-       __renderDiscoveryItem(FROM_FOLLOWED_DOMAIN, 'followedDomainDiscovery', item, index)
-     )}
- 
-     {route.name !== 'Followings' &&
-       unfollowedDomains.length > 0 &&
-       followedDomains.length > 0 && (
-         <View style={styles.unfollowedHeaderContainer}>
-           <Text style={styles.unfollowedHeaders}>{StringConstant.discoveryMoreDomains}</Text>
-         </View>
-       )}
-     {route.name !== 'Followings' &&
-       newMapUnfollowedDomain.map((item, index) =>
-         __renderDiscoveryItem(FROM_UNFOLLOWED_DOMAIN, 'unfollowedDomainDiscovery', item, index)
-       )}
-   </>
-  );
+ const data = isFirstTimeOpen
+   ? [
+       ...followingDomains.map((item, index) => ({
+         ...item,
+         user_id_follower: item.user_id_follower
+       })),
+       {separator: true},
+       ...unfollowingDomains.map((item, index) => ({
+         ...item,
+         user_id_follower: item.user_id_follower
+       }))
+     ]
+   : [...newMapFollowedDomain, {separator: true}, ...newMapUnfollowedDomain];
+ 
+ return (
+   <FlatList
+     onMomentumScrollBegin={handleScroll}
+     contentContainerStyle={{paddingBottom: 100}}
+     data={data}
+     keyExtractor={(_, index) => index.toString()}
+     renderItem={renderItemList}
+     onEndReached={() => fetchData()}
+     onEndReachedThreshold={0.6}
+   />
+ );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->